### PR TITLE
Allow setting http proxy and no proxy environment values

### PR DESCRIPTION
**Pull request by rubroboletus**
**Is this a bug fix or adding new feature?**
new feature

**What is this PR about? / Why do we need it?**
allows this driver to work in VPC behind proxy

**What testing is done?** 
tested in our environment

### CODE REVIEW

**Comment by rubroboletus** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

dear bot, verify

File: charts/aws-ebs-csi-driver/templates/controller.yaml [hunk 1]
```diff
@@ -140,6 +164,14 @@ spec:
           env:
             - name: ADDRESS
               value: /var/lib/csi/sockets/pluginproxy/csi.sock
+{{- if .Values.proxy.http_proxy }}
+            - name: HTTP_PROXY
+              value: {{ .Values.proxy.http_proxy | quote }}
+            - name: HTTPS_PROXY
+              value: {{ .Values.proxy.http_proxy | quote }}
```
Comment (ayberk) [inline charts/aws-ebs-csi-driver/templates/controller.yaml:171]: Hmm would it be better to have a different value for HTTPS_PROXY?

Comment (rubroboletus) [reply charts/aws-ebs-csi-driver/templates/controller.yaml:171]: > Hmm would it be better to have a different value for HTTPS_PROXY?

It depends, in 99% environments is same proxy for both.

Comment (ayberk) [reply charts/aws-ebs-csi-driver/templates/controller.yaml:171]: I think this is a good start. We can do it later if necessary.

Comment (rubroboletus) [reply charts/aws-ebs-csi-driver/templates/controller.yaml:171]: here is lot of ways, how to do it, from my point of view preffered one:
- name: HTTPS_PROXY
  value: {{ .Value.proxy.https_proxy | default .Value.proxy.http_proxy | quote }}

**Comment by ayberk** [comment]
/lgtm
/approve

thanks!

**Pull request merged**